app/controllers/application_controller/buttons.rb

```diff
@@ -100,6 +100,36 @@ def group_update
   # we need a "Provider" so adding an "ExtManagementSystem" to the list.
   MODEL_WITH_OPEN_URL = %w[ExtManagementSystem Service User MiqGroup Tenant CloudTenant GenericObject Vm].freeze
 
+  # Types of visibility for the custom buttons
+  VISIBILITY_TYPES = {'role' => 'role', 'all' => 'all'}.freeze
+
+  def button_visibility_box_edit
+    typ_changed = params[:visibility_typ].present?
+    @edit[:new][:visibility_typ] = VISIBILITY_TYPES[params[:visibility_typ]] if typ_changed
+
+    visibility_typ = @edit[:new][:visibility_typ]
+    if %w[role].include?(visibility_typ)
+      plural = visibility_typ.pluralize
+      key    = plural.to_sym
+      prefix = "#{plural}_"
+
+      @edit[:new][key] = [] if typ_changed
+      params.each do |var, value|
+        next unless var.starts_with?(prefix)
+
+        name = var.split(prefix).last.to_i
+        if value == "1"
+          @edit[:new][key] |= [name] # union
+        elsif value.downcase == "null"
+          @edit[:new][key].delete(name)
```

Comment:
Can you verify it works properly with removal of 1 value or multiple at a time?  I think I did but want to make sure it loops through these properly and clears them all for example if you start with 3 roles and remove 1 or 2 or 3.